### PR TITLE
MatrixRTC: compatibility with new membership format (MSC4143) (ruma api changes)

### PR DESCRIPTION
**Pull request by toger5**
Ruma introduces a slightly different api to be compatible with the new default format for matrix_rtc events.

See: https://github.com/ruma/ruma/pull/1829

- [ ] Public API changes documented in changelogs (optional)

Signed-off-by:

### CODE REVIEW

**Review by bnjbvr — CHANGES_REQUESTED** [review]
(marking "request changes" since this PR is a draft, so as to remove it from my to-be-reviewed list)

**Comment by toger5** [comment]
This still changes the ruma version which is only necessary since the associated ruma PR is not yet merged.

File: Cargo.toml [hunk 1]
```diff
@@ -43,7 +43,7 @@ once_cell = "1.16.0"
 pin-project-lite = "0.2.9"
 rand = "0.8.5"
 reqwest = { version = "0.12.4", default-features = false }
-ruma = { git = "https://github.com/ruma/ruma", rev = "75e8829bec0b7bc5332860e1fb2df658d5c71d66", features = [
+ruma = { git = "https://github.com/toger5/ruma/", rev = "d7094e68e0ed6875f97e0928101d90654d054078", features = [
```
Comment (Hywan) [inline Cargo.toml:46]: Let's wait on https://github.com/ruma/ruma/pull/1829 to be merged before going forward.

File: crates/matrix-sdk-base/src/rooms/normal.rs [hunk 2]
```diff
@@ -1471,6 +1471,7 @@ fn compute_display_name_from_heroes(
 
 #[cfg(test)]
 mod tests {
+    // use crate::rooms::normal::RumaSummary;
```
Comment (bnjbvr) [inline crates/matrix-sdk-base/src/rooms/normal.rs:1474]: nit: commented code here, please remove

File: crates/matrix-sdk-base/src/rooms/normal.rs [hunk 3]
```diff
@@ -2206,6 +2208,8 @@ mod tests {
     async fn test_setting_the_latest_event_doesnt_cause_a_room_info_update() {
         // Given a room,
 
+        use ruma::user_id;
```
Comment (bnjbvr) [inline crates/matrix-sdk-base/src/rooms/normal.rs:2211]: Is this used? if so, can you put it with the regular `use` statements at the top of this module instead?

**Review by bnjbvr — APPROVED** [review]
Thanks! We're trying to use a slightly nicer PR naming convention, https://github.com/matrix-org/matrix-rust-sdk/blob/main/CONTRIBUTING.md#commit-messages-and-pr-title-guidelines can you update the PR title to something shorter that follows this format please?

File: crates/matrix-sdk-base/src/rooms/normal.rs (hunk 3 above)
Comment (toger5) [reply crates/matrix-sdk-base/src/rooms/normal.rs]: Cool that there are guidelines for commit and pr names/messages.

**Review by bnjbvr — APPROVED** [review]

**Pull request merged**